src/components/UI/Controls/DropdownMenu.js

```diff
@@ -64,23 +64,23 @@ const MenuWrapper = styled.div`
 
 function DropdownMenu(props) {
   const [isOpen, setIsOpen] = useState(false);
-  let timeOutId = null;
+  const timeOutId = useRef(null);
 
   useEffect(() => {
-    return function cleanup() {
-      clearTimeout(timeOutId);
+    return () => {
+      clearTimeout(timeOutId.current);
```

Comment:
The problem was that our timeout was cleared before it managed to fire `onBlurHandler`, because the component was re-rendered